### PR TITLE
[DDW-716] Implement static screens for signing dApp interaction transactions

### PR DESCRIPTION
**Pull request by daniloprates**
This PR implements static screens for signing dApp interaction transactions.

## Todos

- [x] dApp transaction request dialog
- [x] Transaction confirmation Assets list component
- [x] Missing token
- [x] Add wallet dropdown option
- [x] Togglable content
- [x] Copy + translation

## Screenshots

screenshot-app zeplin io-2021 08 12-17_52_11

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1628801324059600)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by gabriela-ponce** [comment]
@daniloprates A few comments for this dialog:

- When the amount of the token is not enough, the red tooltip overlaps the other tooltips for that element. Check [this](https://drive.google.com/file/d/1J7n4MGaQgSZMbd2QfiTzjtn4o7zL4bbn/view?usp=sharing) video. Can this be improved? Maybe changing the orientation of the red tooltip?
- There are a few translations missing in Japanese. Check screenshots [1](https://drive.google.com/file/d/1fpV6attkOPYF4wTyfHwGsn8DOhVRBcrU/view?usp=sharing), [2](https://drive.google.com/file/d/1yxNk1-8vg-w0qMCZ7BeL7ZtP-LjNvxRz/view?usp=sharing) and [3](https://drive.google.com/file/d/1cRY8i3GAm4kHOrLR94ldZNnPI0Cu1-6n/view?usp=sharing)
- When there is no enough ADA, there is [no indication on the dialog](https://drive.google.com/file/d/12w-ry1hvFkkWhA29h2YDMVFZT4gBkgF3/view?usp=sharing). Shouldn't the dropdown be outlined in red as on [this](https://app.zeplin.io/project/57acfaa7436a750316161a8b/screen/60ebfaeec23b321391ad1e21) design? Same with the scenario when [the token isn't enough.](https://drive.google.com/file/d/1slgG4fdcKHewCZN4Hkqx0we8BjWN38if/view?usp=sharing)

**Comment by daniloprates** [comment]
@gabriela-ponce thanks! I've fixed the items:


> * When the amount of the token is not enough, the red tooltip overlaps the other tooltips for that element. Check [this](https://drive.google.com/file/d/1J7n4MGaQgSZMbd2QfiTzjtn4o7zL4bbn/view?usp=sharing) video. Can this be improved? Maybe changing the orientation of the red tooltip?

Fixed for [both](https://user-images.githubusercontent.com/1504716/129460488-1c44e464-1fdf-4e94-a1fb-30368ec5507b.png) [cases](https://user-images.githubusercontent.com/1504716/129460490-6e22fb43-7aba-4457-b1f9-0354672a2340.png). Please notice, in case the dialog height is not tall enough, it will [open upwards](https://user-images.githubusercontent.com/1504716/129460521-d5e26b4c-e56f-4260-9918-71e361e252ce.png) and there's nothing we can do.

> * There are a few translations missing in Japanese. Check screenshots [1](https://drive.google.com/file/d/1fpV6attkOPYF4wTyfHwGsn8DOhVRBcrU/view?usp=sharing), [2](https://drive.google.com/file/d/1yxNk1-8vg-w0qMCZ7BeL7ZtP-LjNvxRz/view?usp=sharing) and [3](https://drive.google.com/file/d/1cRY8i3GAm4kHOrLR94ldZNnPI0Cu1-6n/view?usp=sharing)

All fixed ツ

> * When there is no enough ADA, there is [no indication on the dialog](https://drive.google.com/file/d/12w-ry1hvFkkWhA29h2YDMVFZT4gBkgF3/view?usp=sharing). Shouldn't the dropdown be outlined in red as on [this](https://app.zeplin.io/project/57acfaa7436a750316161a8b/screen/60ebfaeec23b321391ad1e21) design? Same with the scenario when [the token isn't enough.](https://drive.google.com/file/d/1slgG4fdcKHewCZN4Hkqx0we8BjWN38if/view?usp=sharing)

It's [done](https://user-images.githubusercontent.com/1504716/129460551-70a9c23a-3621-4115-996e-2be2a184b876.png). I'll reach for proper copy & translation on Monday.

**Comment by dmitrii-gaico** [comment]
@daniloprates Hey Danilo. 

I have a question regarding latest fix for scenario when wallet has not enough ADA. If this happens shouldn't we disable [Continue button](https://drive.google.com/file/d/1bP6A-W1W4Q7rHb7crJAruiwFY-tDT7_E/view?usp=sharing) ? If wallet has not enough ADA then it can't proceed.

**Comment by daniloprates** [comment]
> @daniloprates Hey Danilo.
> 
> I have a question regarding latest fix for scenario when wallet has not enough ADA. If this happens shouldn't we disable [Continue button](https://drive.google.com/file/d/1bP6A-W1W4Q7rHb7crJAruiwFY-tDT7_E/view?usp=sharing) ? If wallet has not enough ADA then it can't proceed.

Nice catch, @dmitrii-gaico. Fixed it.

**Comment by gabriela-ponce** [comment]
@daniloprates All the fixes look great, nice job!

[Here is a screenshot](https://drive.google.com/file/d/18_t2V55YP3Zrwov4gxuaIzgQy_D7Ffwj/view?usp=sharing) of some translations in Japanese that are still missing (I know the first one is still pending, mentioning it for tracking purposes).

**Comment by miorsufianiohk** [comment]
Hi @daniloprates ,

Tested on 18800.

I am not sure if this is relevant or not but I'm including this for consistency reason. I noticed that the displayed index of the token starts from position "0" [screenshot 1](https://drive.google.com/file/d/1YIqKbUHk8AoVj_WI1ZidVKZH9loRa_Zi/view?usp=sharing), in the webApp when sending multiple tokens, it starts from position "1" [screenshot 2](https://drive.google.com/file/d/1QUFutbgDRCj8IZdXbf52EZidmpdCQGLS/view?usp=sharing).

**Comment by daniloprates** [comment]
@gabriela-ponce @miorsufianiohk thanks for the reports. I've added the missing Japanese translation and increased the Token number, so it starts at 1

**Review by gabriela-ponce — APPROVED** [review]

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Great job @daniloprates  👍 . Tested on 18809.

**Review by dmitrii-gaico — APPROVED** [review]

**Comment by alexander-rukin** [comment]
image
something wrong with scroll

**Comment by alexander-rukin** [comment]
image
confused a bit with error message and popup shouldn't be red I suppose

**Comment by alexander-rukin** [comment]
image
image
Errors are about almost the same but one has explanations as a separate sentence, other in brackets

**Comment by alexander-rukin** [comment]
image
looks like this button should be moved to the top (1px)

**Comment by alexander-rukin** [comment]
image
missing spacing between elements

**Comment by alexander-rukin** [comment]
image
image
with 20px (instead of 25px) it looks better

**Comment by alexander-rukin** [comment]
image
Initially, I planned wallet input should be red in any error case, as it's the source of a problem :D any thoughts?

File: source/renderer/app/components/notifications/NotificationActions.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+type Props = {
+  actions: NotificationActionItems,
+};
+
+const NotificationActions = observer(({ actions }: Props) => {
+  const componentStyles = classnames([styles.component]);
+  return (
+    <div className={componentStyles}>
+      {map(actions, (action, key) => {
```
Comment (danielmain) [inline source/renderer/app/components/notifications/NotificationActions.js:33]: Maybe destructuring in line 33 saves you code?  like: "{map(actions, ({className, primary, autoFocus, label, onClick, disabled ....}, key) => {

**Review by danielmain — CHANGES_REQUESTED** [review]
Great Job Danilo. I just added some small semantic corrections as review

**Comment by dmitrii-gaico** [comment]
Not sure if someone already reported a typo so here it is :
image

**Comment by daniloprates** [comment]
> Initially, I planned wallet input should be red in any error case, as it's the source of a problem :D any thoughts?

@alexander-rukin It can be confusing because hovering the wallets dropdown won't give you any feedback

image

**Comment by alexander-rukin** [comment]
> > Initially, I planned wallet input should be red in any error case, as it's the source of a problem :D any thoughts?
> 
> @alexander-rukin It can be confusing because hovering the wallets dropdown won't give you any feedback

it's ok, we highlight error in TX details (we can also consider activating those error dropdown pointing error line while hovering input too) 

oh... btw here we should highlight ada row (as this is cause of an issue) 

image

File: source/renderer/app/components/assets/AssetsTransactionConfirmation.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  assets: Array<AssetToken>,
+  assetsAmounts: Array<BigNumber>,
+  className?: string,
+  adaAmount?: BigNumber,
+  intl: intlShape.isRequired,
+  wallet?: ?Wallet,
+  getAssetByUniqueId: Function,
+};
+
+const AssetsTransactionConfirmation = observer((props: Props) => {
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/assets/AssetsTransactionConfirmation.js:25]: :)

File: source/renderer/app/components/dapp/DappTransactionRequest.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    onAddWallet,
+    onSelectWallet,
+    onSubmit,
+    selectedWallet,
+    transactionFee,
+    triggeredFrom,
+    wallets,
+  } = props;
+  const walletsDropdownError =
+    selectedWallet &&
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/dapp/DappTransactionRequest.js]: I had missed this feature in JS 😱 thanks for taking me out of darkness

**Comment by daniloprates** [comment]
@alexander-rukin @danielmain @dmitrii-gaico thanks for your review and great suggestions. They're done, except the copy changes, which [I've asked](https://input-output-rnd.slack.com/archives/CBECZ2K47/p1629732874016500?thread_ts=1628692075.009300&cid=CBECZ2K47) Nyamh to have a look.

File: source/renderer/app/components/assets/Asset.js [hunk 4]
```diff
@@ -84,6 +84,7 @@ type Props = {
   // In case it's not possible to calculate the container width
   // this props defines after how many characters the `metadata.name` text will cut off
   metadataNameChars?: number,
+  error?: boolean,
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/assets/Asset.js]: Done

File: source/renderer/app/components/assets/AssetContent.js [hunk 5]
```diff
@@ -75,6 +75,7 @@ type Props = {
   highlightFingerprint?: boolean,
   className?: string,
   intl: intlShape.isRequired,
+  error?: boolean,
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/assets/AssetContent.js]: Done

File: source/renderer/app/components/dapp/DappTransactionRequest.js [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    onClose,
+    onAddWallet,
+    onSelectWallet,
+    onSubmit,
+    selectedWallet,
+    transactionFee,
+    triggeredFrom,
+    wallets,
+  } = props;
+  const walletsDropdownError =
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/dapp/DappTransactionRequest.js]: Done

File: source/renderer/app/components/dapp/DappTransactionRequest.js [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+        {intl.formatMessage(messages.fromWalletLabel)}
+      </p>
+      {wallets.length ? (
+        <WalletsDropdown
+          className={walletsDropdownStyles}
+          getStakePoolById={() => {}}
+          numberOfStakePools={100}
+          onChange={onSelectWallet}
+          placeholder={intl.formatMessage(messages.walletsDropdownPlaceholder)}
+          value={selectedWallet ? selectedWallet.id : null}
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/dapp/DappTransactionRequest.js]: Done

File: source/renderer/app/components/notifications/Notification.js [hunk 8]
```diff
@@ -79,6 +85,8 @@ export default class Notification extends Component<Props> {
 
             <div className={messageStyles}>{children}</div>
 
+            {actions && <NotificationActions actions={actions} />}
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/notifications/Notification.js]: Done

File: source/renderer/app/components/notifications/NotificationActions.js (hunk 1 above)
Comment (daniloprates) [reply source/renderer/app/components/notifications/NotificationActions.js]: Done

**Comment by alexander-rukin** [comment]
image
"x" is not centered in the notification

**Comment by alexander-rukin** [comment]
image

you probably got me wrong, I suggested to make ada row red in TX details, not the dropdown

image


it also makes other balances red

**Comment by alexander-rukin** [comment]
image

isn't it possible to have a shift like in designs?

image

p.s. shouldn't it be "Ada" here and not "ADA"?)

**Comment by alexander-rukin** [comment]
image

distance looks too big 

image

here design reference

**Comment by daniloprates** [comment]
> image
> 
> isn't it possible to have a shift like in designs?
> 
> image
> 
> p.s. shouldn't it be "Ada" here and not "ADA"?)

@alexander-rukin what do you mean by "shift"? (For amounts we use the ticker, ADA)

**Comment by alexander-rukin** [comment]
> > image
> > isn't it possible to have a shift like in designs?
> > image
> > p.s. shouldn't it be "Ada" here and not "ADA"?)
> 
> @alexander-rukin what do you mean by "shift"? (For amounts we use the ticker, ADA)
Yes, for amounts "ADA", but for the line name I guess it should be still "ada" or "Ada" (if at the beginning of sentence) 
As for the shift -> look at the vertical line position

**Comment by daniloprates** [comment]
> > > image
> > > isn't it possible to have a shift like in designs?
> > > image
> > > p.s. shouldn't it be "Ada" here and not "ADA"?)
> > 
> > 
> > @alexander-rukin what do you mean by "shift"? (For amounts we use the ticker, ADA)
> 
> Yes, for amounts "ADA", but for the line name I guess it should be still "ada" or "Ada" (if at the beginning of sentence)
> 
> As for the shift -> look at the vertical line position

@alexander-rukin done!

**Comment by alexander-rukin** [comment]
image
in the collapsed view it still lacks margin

**Comment by alexander-rukin** [comment]
image
hover effect here should be same as for the normal icon (increase opacity on hover)

**Comment by alexander-rukin** [comment]
image
scrollbars still look like this

**Comment by alexander-rukin** [comment]
image
not sure what caused this, but right end of a content is shifted to the left. Content should end where button ends

**Comment by daniloprates** [comment]
@alexander-rukin done, except:

> image
> 
> scrollbars still look like this


Which browser are you using? It needs to be Chrome

> image
> 
> not sure what caused this, but right end of a content is shifted to the left. Content should end where button ends

Didn't understand.


This is how it looks to me:






Screen Shot 2021-08-31 at 16 17 12

**Comment by alexander-rukin** [comment]
> Which browser are you using? It needs to be Chrome

I use Chrome Version 92.0.4515.159 (Official Build) (x86_64)

**Comment by gabriela-ponce** [comment]
@daniloprates The fixes look good to me, but the issue that @alexander-rukin mentioned about the scroll is still visible (also using Chrome).
- The scroll bar is only visible when scrolling with the mouse (can't click on it if you don't scroll first). Also, on hover, it looks bigger. Check [this](https://drive.google.com/file/d/1paq6OBi2jaJ9SYdjwpnv5Tt1mYt-p9nI/view?usp=sharing) video.

**Comment by alexander-rukin** [comment]
image
btw I'd expect to see this error message when hovering bottom line in TX (same way as in previous cases)

**Comment by daniloprates** [comment]
> @daniloprates The fixes look good to me, but the issue that @alexander-rukin mentioned about the scroll is still visible (also using Chrome).
> 
> * The scroll bar is only visible when scrolling with the mouse (can't click on it if you don't scroll first). Also, on hover, it looks bigger. Check [this](https://drive.google.com/file/d/1paq6OBi2jaJ9SYdjwpnv5Tt1mYt-p9nI/view?usp=sharing) video.

This is a default behavior for mac. You can change it globally [here](https://user-images.githubusercontent.com/1504716/131686014-c61bb589-0efd-4222-8428-49ca59ef1171.png).

@gabriela-ponce @alexander-rukin I've fixed the different color for the data boxes. They're not as the main dialog. We may need to change the color for all dialog scrollbars, but I prefer to do that in a separated task, testing in the actual app, not only Storybook.

**Comment by daniloprates** [comment]
@alexander-rukin added the Ada error here as well 👇

image

**Comment by alexander-rukin** [comment]
for some reason “computed” line-height here (Ada line)  is 21px -> with 22px it looks better

image

---
all line-heights here 21px
I made all of them 22, so line-height = 22px, box-height = 22px
here on a screenshot look at the alignment:
line - 21px
red box - 22px, but shifted to the top
violet box - 21px and aligned with a text
p.s. left violet box just to see that red is bigger

image


---
btw receiver line-height seems to be 22, as it’s 44 in total ))

image
image

**Comment by daniloprates** [comment]
> for some reason “computed” line-height here (Ada line) is 21px -> with 22px it looks better

> all line-heights here 21px
> I made all of them 22, so line-height = 22px, box-height = 22px

@alexander-rukin done

Screen Shot 2021-09-01 at 17 13 05
Screen Shot 2021-09-01 at 15 37 51

**Comment by alexander-rukin** [comment]
> @alexander-rukin added the Ada error here as well 👇
> 
> image

did you push this?

**Comment by alexander-rukin** [comment]
image

Just noticed that "ADA" suffix has wrong styles - see https://zpl.io/2yDLwD9

**Comment by alexander-rukin** [comment]
image
I can see that scroll bar is not centered on your screenshot 😓

**Review by alexander-rukin — APPROVED** [review]
All good now. @daniloprates you are a star ⭐

**Comment by gabriela-ponce** [comment]
@daniloprates The fixes look good. Just a little comment about [this tooltip](https://user-images.githubusercontent.com/1504716/131697470-e54abad1-480f-4d8d-b299-b74e7ed5cc17.png). Wouldn't it look better to be displayed over the value (100.00000 in this case)? 
I know the current implementation is also how it's done for the tokens, but I think it looks a little odd, as it's not clear what is pointing to.

**Comment by daniloprates** [comment]
> @daniloprates The fixes look good. Just a little comment about [this tooltip](https://user-images.githubusercontent.com/1504716/131697470-e54abad1-480f-4d8d-b299-b74e7ed5cc17.png). Wouldn't it look better to be displayed over the value (100.00000 in this case)?
> I know the current implementation is also how it's done for the tokens, but I think it looks a little odd, as it's not clear what is pointing to.

@gabriela-ponce nice suggestion. The only issue is that, because the Ada Amount element is narrow, it can be opened from the top, as it looks like [this](https://user-images.githubusercontent.com/1504716/131914686-fd98b40d-7e5c-49f6-8969-eb868634878c.png). From the left it works [fine](https://user-images.githubusercontent.com/1504716/131914744-5047e546-f367-4775-8a84-ab46c09cfb73.png). @Alexander-rukin what do you think?

**Comment by alexander-rukin** [comment]
> @gabriela-ponce nice suggestion. The only issue is that, because the Ada Amount element is narrow, it can be opened from the top, as it looks like [this](https://user-images.githubusercontent.com/1504716/131914686-fd98b40d-7e5c-49f6-8969-eb868634878c.png). From the left it works [fine](https://user-images.githubusercontent.com/1504716/131914744-5047e546-f367-4775-8a84-ab46c09cfb73.png). @alexander-rukin what do you think?

I thought we can make the tooltip globally positioned. So it will just go out of bounds for dialog. 

As for now, the tooltip points to the whole row which is an error one. If we want to change position and point exact error part, we will need to do this for every case here. Point amount when it's an amount problem. Point token box when "no such token in wallet" error. But still the whole row should be highlighted in red

**Comment by daniloprates** [comment]
> > @gabriela-ponce nice suggestion. The only issue is that, because the Ada Amount element is narrow, it can be opened from the top, as it looks like [this](https://user-images.githubusercontent.com/1504716/131914686-fd98b40d-7e5c-49f6-8969-eb868634878c.png). From the left it works [fine](https://user-images.githubusercontent.com/1504716/131914744-5047e546-f367-4775-8a84-ab46c09cfb73.png). @alexander-rukin what do you think?
> 
> I thought we can make the tooltip globally positioned. So it will just go out of bounds for dialog.
> 
> As for now, the tooltip points to the whole row which is an error one. If we want to change position and point exact error part, we will need to do this for every case here. Point amount when it's an amount problem. Point token box when "no such token in wallet" error. But still the whole row should be highlighted in red

@gabriela-ponce @alexander-rukin reverted the side positioning

File: source/renderer/app/components/dapp/DappTransactionRequest.js [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+  triggeredFrom: string,
+  wallets: Array<Wallet>,
+};
+
+const DappTransactionRequest = observer((props: Props) => {
+  const [isAdditionalDataVisible, toggleAdditionalData] = useState<boolean>(
+    false
+  );
+  const [isMetadataVisible, toggleMetadata] = useState<boolean>(false);
+  const getToggleLabel = (isVisible: boolean) =>
```
Comment (danielmain) [inline source/renderer/app/components/dapp/DappTransactionRequest.js:102]: Using useCallback you avoid re-defining the function everytime. Additionally its get memoized `const getToggleLabel = useCallback((isVisible: boolean) => { ... }))`

File: source/renderer/app/components/dapp/DappTransactionRequest.js [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+    onSelectWallet,
+    onSubmit,
+    selectedWallet,
+    transactionFee,
+    triggeredFrom,
+    wallets,
+  } = props;
+  let hasAmountError = false;
+  const hasTokenError = assets.reduce((result, token, index) => {
+    if (!selectedWallet) return false;
```
Comment (danielmain) [inline source/renderer/app/components/dapp/DappTransactionRequest.js:125]: Also here, this reduce is called everytime. I would use useCallback or useMemo.  https://reactgo.com/react-usememo-example/

**Review by danielmain — CHANGES_REQUESTED** [review]
I like how you write clean code :) I just added some minor performance Issues when working with functional components. But overall very good!

**Comment by daniloprates** [comment]
> I like how you write clean code :) I just added some minor performance Issues when working with functional components. But overall very good!

Thanks, @danielmain. It's done

**Review by gabriela-ponce — APPROVED** [review]

**Pull request merged**